flang/runtime/unit.cpp

```diff
@@ -149,11 +149,16 @@ std::optional<char32_t> ExternalFileUnit::GetCurrentChar(
     }
     chunk = *recordLength - positionInRecord;
   }
+
+  // Disable buffering if reading from the terminal
```

Comment:
There's always buffering; that's the point of the read/write frames.

Reply:
This reduces the frame size to being only as big as is needed which I think ends up being 1 byte.

Reply:
The frame has to hold the entire record; Fortran FORMAT control descriptors can reposition input and cause input data to be scanned more than once, and obviously we can't re-read data from a terminal or socket or pipe.

Reply:
I guess it does not do anything to the buffer, but it reduces the number of bytes that are read in from doing batch reads to reading in each character one by one and processing them individually.

Reply:
Right, but why reduce the number of bytes that can be read?  The byte count on a (p)read/(p)write system call is a max size, not a minimum requirement.

Reply:
Because otherwise it continues to read until either the frame (or buffer or whatever it is) is full or it reads EOF. So if the program just wants to read in a single integer, ~256 bytes would need to be inputed into the program before the program would get access to that buffer to read in the integer.